### PR TITLE
Snyder-LB133.ag_Costs_USA_C_2005

### PR DESCRIPTION
**Pull request by abigailsnyder**
I tried doing some streamlining, but that ended up making for some very long pipelines. So I tried to compensate with comments. Long story short, really not sure if trying to streamline actually improved it. 

https://github.com/JGCRI/gcamdata/issues/360

### CODE REVIEW

File: inst/extdata/aglu/USDA_crops.csv [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# File: USDA_crops.csv,,
```
Comment (bpbond) [inline inst/extdata/aglu/USDA_crops.csv]: Here and elsewhere, please remove ugly Excel trailing commas at end of header lines

File: R/zchunk_LB133.ag_Costs_USA_C_2005.R [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+#' @details USDA cost information is scaled to the level of GCAM commodity using LDS harvested area and
+#' production data to compute weighting factors for aggregation and convert cost from USD/square meter
+#' to the more useful USD/kg. Cost information is converted to the GCAM base of 1975 USD before aggregation.
+#' For commodities without USDA cost information, the average profit among USDA commodities is combined
+#' with LDS harvested area and production data to calculate cost.
 #' @importFrom assertthat assert_that
 #' @importFrom dplyr filter mutate select
 #' @importFrom tidyr gather spread
-#' @author YourInitials CurrentMonthName 2017
-#' @export
```
Comment (bpbond) [inline R/zchunk_LB133.ag_Costs_USA_C_2005.R]: Remove `@export` line, and rebuild documentation, before final commit.

File: R/zchunk_LB133.ag_Costs_USA_C_2005.R [hunk 3]
```diff
@@ -38,41 +44,258 @@ module_aglu_LB133.ag_Costs_USA_C_2005_DISABLED <- function(command, ...) {
...
+      # have to group by years and store in a dummy value1 column to get the conversion correct:
+      group_by(GCAM_commodity, GTAP_crop, Item, year, value) %>%
+      mutate(value1 = value * gdp_deflator(1975, base_year = year)) %>%
+      # ungroup, drop value and rename value1 to value to get the correct table of 1975 dollars/acre:
+      ungroup() %>% select(-value) %>% rename(value = value1) %>%
+      # Calculate the average across years for each commodity-item combination, and convert to dollars/m2:
+      #   (In particular, the average is computed across non-NA years. So if only 3 of 5 years are non-NA,
+      #   the average is over those 3 numbers, not over 5 with 0's filled in for the NA's.)
+      group_by(GCAM_commodity, GTAP_crop, Item) %>%
+      mutate(cost_75USDm2 = mean(value[!(is.na(value))]) * CONV_M2_ACR) %>%
```
Comment (bpbond) [inline R/zchunk_LB133.ag_Costs_USA_C_2005.R]: Why not just `mean(value, na.rm = TRUE)`?

File: R/zchunk_LB133.ag_Costs_USA_C_2005.R (hunk 2 above)
Comment (abigailsnyder) [inline R/zchunk_LB133.ag_Costs_USA_C_2005.R]: That's so weird, it's done on my local branch. I'll try redoing the documentation, build and reload.

**Comment by bpbond** [comment]
Great–thanks. This is holding on review by @kvcalvin

File: R/zchunk_LB133.ag_Costs_USA_C_2005.R [hunk 4]
```diff
@@ -38,41 +44,258 @@ module_aglu_LB133.ag_Costs_USA_C_2005_DISABLED <- function(command, ...) {
...
+      USDA_cost_data
+
+    # 2. Perform computations
+
+    # Lines 35-67 in original file
+    # Get the cost in 1975 dollars for each GCAM commodity - Item combination covered in the USDA_data_cost table.
+    # First, Take USDA item cost data from input table USDA_cost_data, and add GCAM commodity and GTAP crop mapping info
+    # from the USDA crop mapping input table, USDA_crops.
+    # Then add cost type (variable or na) from the USDA_item_cost input table.
+    # Next, select only vairable price data, only in MODEL_COST_YEARS = 2001:2005 by default
```
Comment (kvcalvin) [inline R/zchunk_LB133.ag_Costs_USA_C_2005.R]: `variable` (not much else to comment on here)

Comment (bpbond) [inline R/zchunk_LB133.ag_Costs_USA_C_2005.R]: Yeah, that'll teach her not to misspell a comment!  :)

Comment (abigailsnyder) [inline R/zchunk_LB133.ag_Costs_USA_C_2005.R]: d'oh XD
I'll do a general spell check...

Comment (abigailsnyder) [inline R/zchunk_LB133.ag_Costs_USA_C_2005.R]: I actually have found a couple more, this was a very helpful reminder!

**Pull request merged**